Update CI macos-debug-qt5 to Xcode 9.3 (and macOS 10.13)

https://github.com/mapbox/mapbox-gl-native/pull/11637 moved CircleCI builds to Xcode 9.3 and macOS 10.13, but the `macos-debug-qt5` build was left behind on Xcode 9.2/macOS 10.12 because of SQLite issues:

```
[ RUN      ] OfflineDatabase.UpdateMetadata
libc++abi.dylib: terminating with uncaught exception of type mapbox::sqlite::Exception: near "id": syntax error Unable to execute statement
make: *** [run-qt-test-*] Abort trap: 6
```

```
$ sqlite3 --version
3.19.3 2017-06-27 16:48:08 2b0954060fe10d6de6d479287dd88890f1bef6cc1beca11bc6cdb79f72e2377b
```

/cc @brunoabinader @tmpsantos

--- a/circle.yml
+++ b/circle.yml
@@ -866,7 +866,7 @@ jobs:
 # ------------------------------------------------------------------------------
   macos-debug-qt5:
     macos:
-      xcode: "9.2.0"
+      xcode: "9.3.0"
     environment:
       BUILDTYPE: Debug
       HOMEBREW_NO_AUTO_UPDATE: 1

